topostats/scars.py

```diff
@@ -0,0 +1,470 @@
+            for k in range(1, max_scar_width + 1):
+                if row + k + 1 >= img.shape[0]:
+                    # Bottom of image, break
+                    break
+                min_scar_value = max(min_scar_value, img[row + k, col])
+                max_border_value = min(img[row, col], img[row + k + 1, col])
+                # Check if scar ended
+                if min_scar_value - max_border_value < -threshold_low * stddev:
+                    while k:
+                        val = (max_border_value - img[row + k, col]) / stddev
+                        marked[row + k, col] = val
+                        k -= 1
+
+    def spread_scars(
+        self,
+        marked: np.ndarray,
+        threshold_low: float,
+        threshold_high: float,
+    ) -> None:
+        """Spread high-marked pixels into adjacent low-marked pixels. This is a smudging function that attempts to catch
+        any pixels that are parts of scars that might not have been extreme enough to get marked above the
+        high_threshold.
+        Any remaining marked pixels below high_threshold are considered not to be scars and are removed from the mask.
+
+        Parameters
+        ----------
+        marked: np.ndarray
+            A 2-D image of pixels that stores the positions of scars marked for removal. The value of the pixel is a
+            floating point value that represents how strongly the algorithm considers it to be a scar.
+            This may or may not already contain non-zero values given that previous iterations of scar removal
+            may have been performed.
+        threshold_low: float
+            A floating point value, that when multiplied by the standard deviation of the image, acts as a
+            threshold for sharp inclines or descents in pixel values and thus marks potential scars.
+            A lower value will make the algorithm
+            more sensitive, and a higher value will make it less sensitive.
+        threshold_high: float
+            A floating point value that is used similarly to threshold_low, however sharp inclines or descents
+            that result in values in the mask higher than this threshold are automatically considered scars.
+
+        Returns
+        -------
+        None
+        """
+
+        # Spread scars that have close to threshold edge-points
+        for row in range(marked.shape[0]):
+            # Spread right
+            for col in range(1, marked.shape[1]):
+                if marked[row, col] >= threshold_low and marked[row, col - 1] >= threshold_high:
+                    marked[row, col] = threshold_high
+
+            # Spread left
+            for col in range(marked.shape[1] - 1, 0):
+                if marked[row, col - 1] >= threshold_low and marked[row, col] >= threshold_high:
+                    marked[row, col - 1] = threshold_high
+
+    def remove_too_short_scars(self, marked: np.ndarray, threshold_high: float, min_scar_length: int) -> None:
```

Comment:
Pedantic but you can ditch the `_too` from this...

```suggestion
    def remove_short_scars(self, marked: np.ndarray, threshold_high: float, min_scar_length: int) -> None:
```